helpers/clean_pnp_backdata.py

```diff
@@ -0,0 +1,320 @@
+import pandas as pd
+import argparse
+import toml
+
+
+def convert_column_datatypes(df):
+    """ Function to convert the column datatypes of a dataframe if they appear
+    within backdata_schema.toml.
+
+    Args:
+        df (pd.DataFrame): The dataframe to convert the column datatypes.
+    Return:
+        DF (pd.DataFrame): The dataframe with the converted column datatypes.
+
+    """
+    with open('./config/backdata_schema.toml', 'r') as f:
+        datatypes_config = toml.load(f)
+
+    datatypes_dict = {}
+
+    for column, datatype in datatypes_config.items():
+        datatypes_dict.update({column: datatype["Deduced_Data_Type"]})
+
+    for i in list(df.columns):
+        if i in list(datatypes_dict.keys()):
+            try:
+                df[i] = df[i].astype(datatypes_dict[i])
+            except:
+                pass
+        else:
+            continue
+
+    return df
+
+
+def get_status_encoded(df):
+    """ Function to populate the statusencoded column based on the mapping of the
+    'map cora status to column "status"' column.
+
+    Args:
+        df (pd.DataFrame): The dataframe to populate the statusencoded column.
+    Return:
+        DF (pd.DataFrame): The dataframe with the populated statusencoded column.
+    """
+    cora_to_status_encoded_dict = {200: 100,
+                                   100: 101,
+                                   1000: 102,
+                                   400: 200,
+                                   500: 201,
+                                   600: 210,
+                                   800: 211,
+                                   1200: 302,
+                                   1300: 303,
+                                   900: 304,
+                                   1400: 309}
+
+    df["statusencoded"] = df['map cora status to column "status"'].map(
+        cora_to_status_encoded_dict
+    )

```

Comment:
Hi Ryan,
this is a bit more complicated, you are correct with your mapping of cora to statusencoded, but you need to update the status column. Have a look in one of our intermediate output files for how to do this.

(create the "status" column by mapping 210 and 211 to "Clear" and "Clear - overridden" - check in the outputs.)

Reply:
I have addressed your comment with my latest commit. Thanks.